### PR TITLE
helper functions for handling k8s objects

### PR DESCRIPTION
**Pull request by ShwethaKumbla**
This PR adds additional helper methods to handle k8s objects.

### CODE REVIEW

File: klient/k8s/resources/resources.go [hunk 1]
```diff
@@ -121,3 +133,49 @@ func (r *Resources) List(ctx context.Context, objs k8s.ObjectList, opts ...ListO
...
+	p := cr.RawPatch(patch.PatchType, patch.Data)
+
+	o := &cr.PatchOptions{Raw: patchOptions}
+	return r.client.Patch(ctx, objs, p, o)
+}
+
+// BuildFromJSON encodes a JSON string as a typed (or unstructured) object representation.
+// TODO: needs to define the object structure to full fill the functionality
+func (r *Resources) BuildFromJSON(obj k8s.Object, jsonData string) error {
+	return json.Unmarshal([]byte(jsonData), &obj)
```
Comment (vladimirvivien) [inline klient/k8s/resources/resources.go:170]: I don't think we should use the plain JSON encoder.
Instead, look in the [api-machinery runtime for tests on how to use encoders](https://sourcegraph.com/github.com/kubernetes/apimachinery@master/-/blob/pkg/runtime/embedded_test.go)

File: klient/k8s/resources/resources.go [hunk 2]
```diff
@@ -121,3 +133,49 @@ func (r *Resources) List(ctx context.Context, objs k8s.ObjectList, opts ...ListO
 	o := &cr.ListOptions{Raw: listOptions}
 	return r.client.List(ctx, objs, o)
 }
+
+func WithLabelSelector(sel string) ListOption {
```
Comment (vladimirvivien) [inline klient/k8s/resources/resources.go:137]: Love seeing these convenience functions for the options!!!

File: klient/k8s/resources/resources.go (hunk 1 above)
Comment (vladimirvivien) [reply klient/k8s/resources/resources.go:170]: Actually this package is better and it applies the proper object conversions https://sourcegraph.com/github.com/kubernetes/apimachinery@master/-/tree/pkg/util/json

**Review by vladimirvivien — COMMENTED** [review]
This is awesome work. I left comment on using the runtime json converters instead of plain json encoder. See https://sourcegraph.com/github.com/kubernetes/apimachinery@master/-/tree/pkg/util/json

File: klient/k8s/resources/resources.go (hunk 1 above)
Comment (ShwethaKumbla) [reply klient/k8s/resources/resources.go:170]: Just marked this PR to WIP to update the changes for JSON encoding .

Comment (ShwethaKumbla) [reply klient/k8s/resources/resources.go]: JSON encoding will be taken care as part of #43

**Review by vladimirvivien — APPROVED** [review]
lgtm

**Pull request merged**